src/aio.rs

```diff
@@ -145,7 +250,46 @@ where
     Ok(())
 }
 
-async fn connect_simple(connection_info: &ConnectionInfo) -> RedisResult<ActualConnection> {
+async fn connect_simple_tokio(connection_info: &ConnectionInfo) -> RedisResult<ActualConnection> {
+    Ok(match *connection_info.addr {
+        ConnectionAddr::Tcp(ref host, port) => {
+            let socket_addr = {
+                let mut socket_addrs = (&host[..], port).to_socket_addrs()?;
+                match socket_addrs.next() {
+                    Some(socket_addr) => socket_addr,
+                    None => {
+                        return Err(RedisError::from((
+                            ErrorKind::InvalidClientConfig,
+                            "No address found for host",
+                        )));
+                    }
+                }
+            };
+
+            TcpStreamTokio::connect(&socket_addr)
```

Comment:
I think we could remove some duplication by adding a private trait to abstract over connection.

```rust
trait Connect {
    fn connect_tcp(socket_addr: SocketAddr) -> RedisResult<ActualConnection>;
    fn connect_udp(path: &Path) -> RedisResult<ActualConnection>;
}
struct Tokio;
impl Connect for Tokio {}
struct AsyncStd;
impl Connect for AsyncStd {}
```

What do you think?

Reply:
Yeah, lets do it. I assume that you mean "unix" instead of "udp". right?